### PR TITLE
chore(upstream-updates): update app from upstream

### PR DESCRIPTION
**Pull request by gminn**
### Summary

Update from upstream.

Used:

```
diff -duwr memfault-asset-tracker/ nrf/applications/asset_tracker_v2/ | patch -p0
```

Resolved a few rejected patch changes, and reverted some changes that removed features we've added.

### Test Plan

- Flashed thingy91, uploaded data
- CI

### CODE REVIEW

**Comment by gminn** [comment]
* **#41** <a href="https://app.graphite.dev/github/pr/memfault/memfault-asset-tracker/41?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gminn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: overlay-memfault.conf [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 CONFIG_MEMFAULT=y
 CONFIG_MEMFAULT_NCS_PROJECT_KEY=""
+CONFIG_MEMFAULT_NCS_DEVICE_ID_IMEI=y
```
Comment (gminn) [inline overlay-memfault.conf:9]: Note this is already the default, nordic just added to make sure their CI built: https://github.com/memfault/sdk-nrf/commit/489a528d7fcc46982377822448c37ce4ad55309f

File: configuration/memfault/memfault_metrics_heartbeat_config.def [hunk 2]
```diff
@@ -2,7 +2,7 @@
  * Please refer to https://docs.memfault.com/docs/embedded/metrics-api for more details.
  */
 
-MEMFAULT_METRICS_KEY_DEFINE(GnssTimeToFix, kMemfaultMetricType_Unsigned)
-MEMFAULT_METRICS_KEY_DEFINE(GnssSatellitesTracked, kMemfaultMetricType_Unsigned)
-MEMFAULT_METRICS_KEY_DEFINE(LocationTimeoutSearchTime, kMemfaultMetricType_Unsigned)
+MEMFAULT_METRICS_KEY_DEFINE(gnss_time_to_fix_ms, kMemfaultMetricType_Unsigned)
```
Comment (noahp) [inline configuration/memfault/memfault_metrics_heartbeat_config.def:5]: 🥹

File: doc/cloud_module.rst [hunk 3]
```diff
@@ -95,7 +95,7 @@ This enables the cloud to issue FOTA updates and update the application and mode
 For additional documentation on the various FOTA implementations, refer to the respective client library documentation linked to in :ref:`Integration layers <integration_layers>`.
 
 Full modem FOTA updates are only supported by nRF Cloud.
-This application implements full modem FOTA only for the nRF9161 DK, and for nRF9160 DK version 0.14.0 and higher.
+This application implements full modem FOTA for the nRF91x1 DKs and for the nRF9160 DK version 0.14.0 and higher.
```
Comment (noahp) [inline doc/cloud_module.rst:98]: lol, bit aspirational there nordic

**Review by noahp — APPROVED** [review]
sweet! lgtm

**Pull request merged**